server/meca/file-generators/coverLetter.test.js

```diff
@@ -1,16 +1,39 @@
-const coverLetter = require('./coverLetter')
+const PDFParser = require('pdf2json')
+const generateCoverLetter = require('./coverLetter')
 
-describe('Cover letter HTML generator', () => {
-  it('returns HTML document', async () => {
-    const html = await coverLetter({})
-    expect(html).toContain('<!DOCTYPE html>')
+const coverLetter = "<h1>Pick Me!</h1><p>I'm simply the best</p>"
+
+describe('Cover letter PDF generator', () => {
+  it('returns PDF document', async () => {
+    const pdf = await generateCoverLetter({
+      coverLetter,
+    })
+    const pdfString = pdf.toString()
+    expect(pdfString.indexOf('%PDF')).toBe(0)
   })
 
-  test.each([['coverLetter']])('replaces %s', async key => {
-    const randomString = Math.random()
-      .toString(36)
-      .substr(2)
-    const html = await coverLetter({ [key]: randomString })
-    expect(html).toContain(randomString)
+  it('PDF document contains the cover letter', async () => {
+    const document = await generateCoverLetter({
+      coverLetter,
+    })
+
+    const pdfParser = new PDFParser()
+    let errors = 0
+    pdfParser.on('pdfParser_dataError', err => {
+      if (err) errors += 1
+    })
+
+    expect(() => pdfParser.parseBuffer(document)).not.toThrow()
+    pdfParser.on('pdfParser_dataReady', pdfData => {
+      expect(errors).toBe(0)
+      expect(pdfParser.data).not.toBe(null)
+
+      const text = pdfParser.data.Pages[0].Texts.map(item =>
+        item.R.map(t => t.T).reduce((prev, curr) => prev + curr),
+      ).reduce((prev, curr) => prev + curr)
+
+      const jsonData = JSON.stringify(text, null, 2)
+      expect(jsonData).toBe('"%20C%20overLetterPickMe!I\'msimplythebest"')
```

Comment:
Does the following work? 
https://nodejs.org/api/buffer.html#buffer_buf_includes_value_byteoffset_encoding
```suggestion
      expect(document.includes('"%20C%20overLetterPickMe!I\'msimplythebest"')).toBe(true)
```

Reply:
`jsonData` is a string because of `stringify` so I think this should be okay no?

Reply:
yeah the pdf2json library decodes the PDF stream which is not as easy as you might think. Then the `map-map-reduce-reduce` has to do some funky things to get the text out of the json.